index.js

```diff
@@ -9,7 +9,216 @@ try {
   }
 }
 
-const {Parser, Node, NodeArray, TreeCursor, pointTransferArray} = binding;
+const {Parser, NodeMethods, Tree, TreeCursor} = binding;
+
+const {rootNode} = Tree.prototype;
+
+Object.defineProperty(Tree.prototype, 'rootNode', {
+  get() {
+    if (!this.nodes) this.nodes = {};
+    rootNode.call(this);
+    return unmarshalNode(this);
+  }
+})
+
+class SyntaxNode {
+  constructor(tree) {
+    this.tree = tree;
+  }
```

Comment:
This class should replace `Node` in the `.d.ts` file right? Along with all it's info